store/tikv/2pc.go

```diff
@@ -100,14 +99,32 @@ func newTwoPhaseCommitter(txn *tikvTxn) (*twoPhaseCommitter, error) {
 				Key: lockKey,
 			}
 			keys = append(keys, lockKey)
+			size += len(lockKey)
+		}
+	}
+	txnWriteKVCountHistogram.Observe(float64(len(keys)))
+	txnWriteSizeHistogram.Observe(float64(size / 1024))
+
+	// Increase lockTTL for large transactions.
+	// The algorithm is simply linearly increasing according to the size of the
+	// data to be written. The lockTTL will reach to maxLockTTL(120s) at a
+	// write size of 20MB, approximately equivalent to removing about 1 million
+	// keys.
+	var lockTTL uint64
+	if size > txnCommitBatchSize {
+		lockTTL = uint64(float64(defaultLockTTL) * float64(size) / float64(txnCommitBatchSize))
```

Comment:
I think the lockTTL should not take `txnCommitBatchSize` into account, because we may change the batch size in the future.
We should use a constant specifically for lock TTL.